service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/RemoveOrderController.java

```diff
@@ -12,84 +12,77 @@
         Map<String, Object> data = request.getCaseDetails().getData();
-        CaseData caseData = mapper.convertValue(data, CaseData.class);
+        CaseData caseData = getCaseData(request.getCaseDetails());
 
-        data.put(REMOVABLE_ORDER_LIST_KEY, service.buildDynamicListOfOrders(caseData.getOrderCollection()));
+        data.put(REMOVABLE_ORDER_LIST_KEY, service.buildDynamicListOfOrders(caseData));
 
-        return AboutToStartOrSubmitCallbackResponse.builder()
-            .data(data)
-            .build();
+        return respond(data);
     }
 
     @PostMapping("/mid-event")
     public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest request) {
-        Map<String, Object> data = request.getCaseDetails().getData();
-        CaseData caseData = mapper.convertValue(data, CaseData.class);
+        CaseDetails caseDetails = request.getCaseDetails();
+        CaseDetailsMap caseDetailsMap = caseDetailsMap(caseDetails);
+        CaseData caseData = getCaseData(caseDetails);
 
         // When dynamic lists are fixed this can be moved into the below method
-        UUID id = getDynamicListSelectedValue(caseData.getRemovableOrderList(), mapper);
+        UUID removedOrderId = getDynamicListSelectedValue(caseData.getRemovableOrderList(), mapper);
+        RemovableOrder removableOrder = service.getRemovedOrderByUUID(caseData, removedOrderId);
 
-        data.putAll(service.populateSelectedOrderFields(caseData.getOrderCollection(), id));
+        service.populateSelectedOrderFields(caseData, caseDetailsMap, removedOrderId, removableOrder);
 
         // Can be removed once dynamic lists are fixed
-        data.put(REMOVABLE_ORDER_LIST_KEY, service.buildDynamicListOfOrders(caseData.getOrderCollection(), id));
+        caseDetailsMap.put(REMOVABLE_ORDER_LIST_KEY, service.buildDynamicListOfOrders(caseData, removedOrderId));
 
-        return AboutToStartOrSubmitCallbackResponse.builder()
-            .data(data)
-            .build();
+        return respond(caseDetailsMap);
     }
 
     @PostMapping("/about-to-submit")
     public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest request) {
         CaseDetails caseDetails = request.getCaseDetails();
-        Map<String, Object> data = caseDetails.getData();
-        CaseData caseData = mapper.convertValue(data, CaseData.class);
-
-        List<Element<GeneratedOrder>> orders = caseData.getOrderCollection();
-        List<Element<GeneratedOrder>> hiddenOrders = caseData.getHiddenOrders();
+        CaseDetailsMap caseDetailsMap = caseDetailsMap(caseDetails);
+        CaseData caseData = getCaseData(caseDetails);
 
-        data.put("children1", service.removeFinalOrderPropertiesFromChildren(caseData));
+        UUID removedOrderId = getDynamicListSelectedValue(caseData.getRemovableOrderList(), mapper);
+        RemovableOrder removableOrder = service.getRemovedOrderByUUID(caseData, removedOrderId);
```

Comment:
I would move this into the service, controller doesn't need to do this. Same in the mid event